code/game/objects/items/weapons/tools.dm

```diff
@@ -576,3 +577,82 @@
 		open = !open
 		user << "<span class='notice'>You [open?"open" : "close"] the conversion kit.</span>"
 		update_icon()
+
+/*
+ * Soldering Iron
+ */
+/obj/item/weapon/solder
+	name = "soldering iron"
+	icon = 'icons/obj/items.dmi'
+	icon_state = "solder-0"
+	hitsound = 'sound/weapons/toolhit.ogg'
+	flags = FPRINT
+	siemens_coefficient = 1
+	slot_flags = SLOT_BELT
+	force = 3.0
+	sharpness = 1
+	throwforce = 5.0
+	throw_speed = 1
+	throw_range = 5
+	w_class = 2.0
+	m_amt = 70
+	g_amt = 30
+	w_type = RECYK_MISC
+	melt_temperature = MELTPOINT_PLASTIC
+	origin_tech = "engineering=1"
+
+	var/status = 1 		//Whether the welder is secured or unsecured (able to attach rods to it to make a flamethrower)
+	var/max_fuel = 20 	//The max amount of fuel the welder can hold
+
+/obj/item/weapon/solder/New()
+	. = ..()
+	create_reagents(max_fuel)
+	//Does not come fueled up
+
+/obj/item/weapon/solder/update_icon()
+	..()
+	if(reagents.get_reagent_amount("sacid")>15)

```

Comment:
I'd prefer you use a switch here, since it looks a bit neater and requires less proc calls.
Consider

    switch(reagents.get_reagent_amount("sacid"))
        if(15 to INFINITY)
            yadda yadda
        if(10 to 15)
            etc.